src/nunit.analyzers/ConstActualValueUsage/ConstActualValueUsageAnalyzer.cs

```diff
@@ -24,90 +22,74 @@ public class ConstActualValueUsageAnalyzer : BaseAssertionAnalyzer
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(descriptor);
 
-        protected override void AnalyzeAssertInvocation(SyntaxNodeAnalysisContext context,
-            InvocationExpressionSyntax assertExpression, IMethodSymbol methodSymbol)
+        protected override void AnalyzeAssertInvocation(OperationAnalysisContext context, IInvocationOperation assertOperation)
         {
-            bool IsLiteralExpression(ExpressionSyntax expression)
+            static bool IsLiteralOperation(IOperation operation)
             {
-                if (expression is LiteralExpressionSyntax)
+                if (operation is ILiteralOperation)
                     return true;
 
-                if (expression is PrefixUnaryExpressionSyntax prefixUnaryExpression)
-                    return IsLiteralExpression(prefixUnaryExpression.Operand);
+                if (operation is IUnaryOperation unary)
+                    return IsLiteralOperation(unary.Operand);
 
-                if (expression is BinaryExpressionSyntax binaryExpression)
-                    return IsLiteralExpression(binaryExpression.Left) && IsLiteralExpression(binaryExpression.Right);
-
-                if (expression is ParenthesizedExpressionSyntax parenthesizedExpression)
-                    return IsLiteralExpression(parenthesizedExpression.Expression);
+                if (operation is IBinaryOperation binary)
+                    return IsLiteralOperation(binary.LeftOperand) && IsLiteralOperation(binary.RightOperand);
 
                 return false;
             }
 
-            bool IsConstant(ExpressionSyntax expression)
-            {
-                if (IsLiteralExpression(expression))
-                    return true;
-
-                var argumentSymbol = context.SemanticModel.GetSymbolInfo(expression).Symbol;
-
-                return (argumentSymbol is ILocalSymbol localSymbol && localSymbol.IsConst) ||
-                       (argumentSymbol is IFieldSymbol fieldSymbol && fieldSymbol.IsConst);
-            }
-
-            bool IsStringEmpty(ExpressionSyntax expression)
+            static bool IsStringEmpty(IOperation operation)
             {
-                return expression is MemberAccessExpressionSyntax memberAccessExpression &&
-                    string.Equals(memberAccessExpression.Expression.ToString(), "string", StringComparison.OrdinalIgnoreCase) &&
-                    string.Equals(memberAccessExpression.Name.Identifier.Text, "Empty", StringComparison.Ordinal);
+                return operation is IFieldReferenceOperation propertyReference
+                    && propertyReference.Field.Name == nameof(string.Empty)
+                    && propertyReference.Field.ContainingType.SpecialType == SpecialType.System_String;
             }
 
-            void Report(ExpressionSyntax expression)
+            void Report(SyntaxNode expression)
             {
```

Comment:
Change parameter to IOperation. This then moved the .Syntax call from the callers to the method.